phoenix-core/src/main/java/org/apache/phoenix/exception/DataExceedsCapacityException.java

```diff
@@ -31,11 +31,26 @@ public DataExceedsCapacityException(String message) {
                 SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY).setMessage(message).build().buildException());
     }
     
+    @Deprecated
     public DataExceedsCapacityException(PDataType type, Integer precision, Integer scale, String columnName, ImmutableBytesWritable value) {
         super(new SQLExceptionInfo.Builder(SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY)
             .setMessage((columnName == null ? "" : columnName + " ") + getTypeDisplayString(type, precision, scale, value))
             .build().buildException());
     }
+
+    public DataExceedsCapacityException(String columnName, String mesage) {
+        super(new SQLExceptionInfo.Builder(
+                SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY)
+                .setColumnName(columnName)
+                .setMessage(mesage).build()
+                .buildException());
+    }
+
+    public DataExceedsCapacityException(String columnName, int columnLength) {
+        this(columnName, "maxLength=" + columnLength);
+    }
+
+    @Deprecated
```

Comment:
You could also deprecate the method `private static String getTypeDisplayString(PDataType type, Integer precision, Integer scale, ImmutableBytesWritable value)` below to avoid accidental usage in future, and add a new helper method that doesn't take `value` as the argument.